### DISCUSSION TITLE
Transitive dependency installation with no-quarantine flag

### DISCUSSION THREAD

**Discussion by marcprux** [Homebrew/discussions#2328] [Writing Formulae/Casks] [answered]
I have an app that needs to be installed with:

```
brew install --no-quarantine --cask org/casks/secondary-app
```

The formula has a `depends_on cask: "primary-app"`, but when the secondary-app installs the primary-app, the `--no-quarantine` flag is not transitively applied, and the "primary-app" cannot be launched without the usual gatekeeper bypass right-click:

```
==> Tapping org/casks
Cloning into '/usr/local/Homebrew/Library/Taps/org/homebrew-casks'...
remote: Enumerating objects: 51, done.
remote: Counting objects: 100% (51/51), done.
remote: Compressing objects: 100% (40/40), done.
remote: Total 51 (delta 8), reused 49 (delta 6), pack-reused 0
Receiving objects: 100% (51/51), 8.35 KiB | 4.17 MiB/s, done.
Resolving deltas: 100% (8/8), done.
==> Installing Cask primary-app
==> Moving App 'Primary App.app' to '/Applications/Primary App.app'
🍺  primary-app was successfully installed!
==> Installing Cask secondary-app
Warning: macOS's Gatekeeper has been disabled for this Cask
==> Moving App 'Secondary App.app' to '/Applications/Secondary App.app'
🍺  secondary-app was successfully installed!
```

Is there any way to permit the `--no-quarantine` flag to be applied transitively to dependencies that it installs? I'm hoping for something like this:

```ruby
cask "secondary-app" do
  name "Secondary App"
  homepage "https://github.com/user/secondard-app/"
  app "Secondary App.app"
  depends_on cask: "primary-app", args: { no_quarantine: true }
  version "1.2.3"
  url "https://github.com/user/secondary-app/releases/download/#{version}/SecondaryApp.zip"
  sha256 "9f833e270826ac40b9d40416d202238767b9bbf9d598c1aa2558af318b4cb9f1"
end                                                                   
```

**Comment by SMillerDev** [comment, marked as answer]
Definitely not in the cask file since the command is a modification on the install procedure.

Setting it in [`HOMEBREW_CASK_OPTS`](https://docs.brew.sh/Manpage#environment) will achieve this result though

**Comment by marcprux** [comment]
Thanks for the response. I'm fairly new to the system, but my understand is that `HOMEBREW_CASK_OPTS` is something that the user needs to set explicitly in their configuration, correct? It can't be used as a means to automatically pass the `no-quarantine` behavior transitively through to dependencies, right?

**Reply by SMillerDev** [reply]
No, because `--no-quarantine` is something specific to the user install and explicitly disables a security feature so we don't want to set that in homebrew.
